packages/openapi-generator/src/wrapper-files/tsconfig-json.spec.ts

```diff
@@ -0,0 +1,22 @@
+import { resolve } from 'path';
+import { readJSON } from '@sap-cloud-sdk/util';
+import { GeneratorOptions } from '../options';
+import { tsconfigJson } from './tsconfig-json';
+
+describe('tsconfigJson', () => {
+  it('contains the needed dom lib', () => {
+    const tsConfig = JSON.parse(tsconfigJson({} as GeneratorOptions));
+    expect(tsConfig.compilerOptions.lib).toContain('dom');
+  });
+
+  it('returns the custom ts config content', () => {
```

Comment:
This test will actually do fs and relies on a directory that is not part of this package. The fs-mock package that Tom introduced is perfect to mock this away and provide some dummy file content. I think this would be the better maintainable and more performant approach.

Reply:
Then it would be symmetric to the OData case

Reply:
I agree that this is rather an E2E test.
And you are right, I think execa would require different handling if combined with mock-fs. Then again, execa is has nothing to do with the module at hand (tsconfig-json.ts) and therefore should not be of concern for this spec.
I don't fully understand the dist dir thing, but let's discuss this separately.